schemas/com.snowplowanalytics.snowplow/recoveries/jsonschema/1-0-0

```diff
@@ -0,0 +1,145 @@
+            "enum": [
+              "RemoveFromQueryString",
+              "RemoveFromBody"
+            ]
+          },
+          "error": {
+            "description": "Error filter used to discriminate the bad rows that will be affected by this recovery scenario",
+            "type": "string"
+          },
+          "toRemove": {
+            "description": "String or regex to remove from the {body, query string}",
+            "type": "string"
+          }
+        },
+        "required": [
+          "name",
+          "error",
+          "toRemove"
+        ],
+        "additionalProperties": false
+      },
+      {
+        "properties": {
+          "name": {
+            "description": "The ReplaceIn{Body,QueryString} will replace part of the payload",
+            "enum": [
+              "ReplaceInQueryString",
+              "ReplaceInBody"
+            ]
+          },
+          "error": {
+            "description": "Error filter used to discriminate the bad rows that will be affected by this recovery scenario",
+            "type": "string"
+          },
+          "toReplace": {
+            "description": "String or regex to replace in the {body, query string}",
+            "type": "string"
+          },
+          "replacement": {
+            "description": "Replacement for the {body, query string}",
+            "type": "string"
+          }
+        },
+        "required": [
+          "name",
+          "error",
+          "toReplace",
+          "replacement"
+        ],
+        "additionalProperties": false
+      },
+      {
+        "properties": {
+          "name": {
+            "description": "The ReplaceInBase64FieldIn{Body,QueryString} will replace part of a base64-encoded field",
+            "enum": [
+              "ReplaceInBase64FieldInQueryString",
+              "ReplaceInBase64FieldInBody"
+            ]
+          },
+          "error": {
+            "description": "Error filter used to discriminate the bad rows that will be affected by this recovery scenario",
+            "type": "string"
+          },
+          "base64Field": {
+            "description": "Name of the base64-encoded field to decoded and do a replacement in",
+            "enum": [
+              "ue_px",
+              "cx"
+            ]
+          },
+          "toReplace": {
+            "description": "String or regex to replace in the specified base64-encoded field in the {body, query string}",
+            "type": "string"
+          },
+          "replacement": {
+            "description": "Replacement in the specified base64-encoded field in the {body, query string}",
+            "type": "string"
+          }
+        },
+        "required": [
+          "name",
+          "error",
+          "toReplace",
+          "replacement"
+        ],
+        "additionalProperties": false
+      },
+      {
+        "properties": {
```

Comment:
I'd still prefer an option with self-describing JSON as a fallback. `additionalProperties` makes this fallback effectively untyped.

Reply:
The point is not to have to upload a schema to run a custom recovery scenario, I think having a self-desc fallback defeats this purpose, wdyt?

Reply:
At one hand - yes, definitely, but on the other hand if somebody wrote a custom recovery scenario - uploading a schema would be a least of their problems. At the same time schema'ed recovery can motivate one to publish whole recovery scenario.

Also, even if recovery will be written in Scala - JSON schema has stronger validation, so quite a lot invalid configs can be filtered out.

Reply:
I'm not quite sure adding steps is going to encourage contributions but anyways the main problem I'm seeing with this approach is that I'm not going to have the same ADT in the `data` array, that seems like unnecessary complexity, wdyt?